gui/wxpython/dbmgr/dialogs.py

```diff
@@ -222,10 +222,13 @@ def GetSQLString(self, updateValues=False):
                     ctype = columns[name]["ctype"]
                     value = columns[name]["values"][idx]
                     id = columns[name]["ids"][idx]
-                    try:
-                        newvalue = self.FindWindowById(id).GetValue()
-                    except:
-                        newvalue = self.FindWindowById(id).GetLabel()
+                    widget = self.FindWindowById(id)
+                    if hasattr(widget, "GetValue"):
+                        newvalue = widget.GetValue()
+                    elif hasattr(widget, "GetLabel"):
+                        newvalue = widget.GetLabel()
+                    else:
+                        raise AttributeError(f"AttributeError in {widget}")
```

Comment:
I think I would prefer a solution closer to the original:

```suggestion
                    widget = self.FindWindowById(id)
                    if hasattr(widget, "GetValue"):
                        newvalue = widget.GetValue()
                    else:
                        newvalue = widget.GetLabel()
```

Reply:
Would we not require to handle the case where `id` isn't found? Something where neither `GetValue()` or `GetLabel()` works?

Reply:
I think that would be an unexpected bug and in that case we want to know about it and let it throw the exception in the same way it would do it before this PR.

Reply:
Yeah, thats basically what I had in mind, maybe sth like this:
```
try:
    widget = self.FindWindowById(id)
    if hasattr(widget, "GetValue"): # check for each of the values
        newvalue = widget.GetValue()
    else:
        newvalue = widget.GetLabel()
except Exception as e: # the unexpected case where object is not found
    print(e)
```

Reply:
I think the original code is basically right, it uses "ask forgiveness than permission (EAFP)" as opposed to "look before you leap (LBYL)". It just needs to be little more careful about exceptions caught. The following does FindWindowById outside of the try-except, then it tries one attribute and in except another. Attribute error on the second one will be propagated from except and I think Python will mention also the original exception (from try portion) in a standard traceback.

```suggestion
                    widget = self.FindWindowById(id)
                    try:
                        newvalue = widget.GetValue()
                    except AttributeError:
                        newvalue = widget.GetLabel()
```

I don't see anything wrong with the above code as long as we say we use EAFP. @petrasovaa suggestion is switching to LBYL which would be fine too. I don't see an advantage in combining them together here. We want the exception to be raised at the end if the attribute is not that - I mean that's what the intention of original code seems to be (it doesn't do anything to mitigate the exception for the GetLabel case).